pkg/cli/admin/upgrade/upgrade.go

```diff
@@ -340,15 +347,12 @@ func (o *Options) Run() error {
 		return nil
 
 	default:
-		if c := findClusterOperatorStatusCondition(cv.Status.Conditions, configv1.OperatorDegraded); c != nil && c.Status == configv1.ConditionTrue {
-			prefix := "No upgrade is possible due to an error"
-			if c := findClusterOperatorStatusCondition(cv.Status.Conditions, configv1.OperatorProgressing); c != nil && c.Status == configv1.ConditionTrue && len(c.Message) > 0 {
-				prefix = c.Message
-			}
-			if len(c.Message) > 0 {
-				return fmt.Errorf("%s:\n\n  Reason: %s\n  Message: %s\n\n", prefix, c.Reason, strings.ReplaceAll(c.Message, "\n", "\n  "))
+		if c := findClusterOperatorStatusCondition(cv.Status.Conditions, clusterStatusFailing); c != nil {
+			if c.Status != configv1.ConditionFalse {
+				fmt.Fprintf(o.Out, "%s=%s:\n\n  Reason: %s\n  Message: %s\n\n", c.Type, c.Status, c.Reason, strings.ReplaceAll(c.Message, "\n", "\n  "))
 			}
-			return fmt.Errorf("The cluster can't be upgraded, see `oc describe clusterversion`")
+		} else {
+			fmt.Fprintf(o.ErrOut, "warning: No current %s info, see `oc describe clusterversion` for more details.\n", clusterStatusFailing)
```

Comment:
Because we have removed the return statement , the upgrade might be able to start with the warning?

Reply:
those `return` were just for the dead-code `Degraded` processing, that never did anything, because we never have `Degraded` in ClusterVersion.  Previously we ignored `Failing` completely.  Now we'll tell you about it, and if you're asking for an update, we'll ask you to set `--allow-upgrade-with-warnings`.